dedupUMI

goeringr opened the pull request:
Including option to deduplicate UMIs as a flag in both AlignUMIquant and Pigpen. This requires reads to have UMIs appended to the read header by UMI_tools extract

@taliaferrojm, Take a look!

goeringr committed:
dedupUMI

Including option to deduplicate UMIs as a flag in both AlignUMIquant and Pigpen. This requires reads to have UMIs appended to the read header by UMI_tools extract

taliaferrojm commented:
I think this is mostly good, but from what I can see, if you use dedup, the list of bams gets renamed as samplebams. If not, it stays as starbams. This leads to some messy if/else statements later which are kinda unneccesary because the parameters used in those downstream steps are the same whether or not you used dedup.

Is it easier and cleaner to, at the end of the deduplication function in `pigpen.py`, rename that list of bams as starbams? Then nothing much downstream needs to change, at least in terms of variable names.

goeringr committed:
more UMI tools updates

simple script for extracting UMIs and simplification of --UMIdedup flag in pigpen.py

goeringr committed:
added --libType parameter

This will make alignUMIquant.py more single amplicon friendly. Unneccessary proccesses (salmon, postmaser) will not be ran for single amplicon libraries.
